### DISCUSSION TITLE
Allow styling through @emotion/styled

### DISCUSSION THREAD

**Discussion by jembach** [mantinedev/mantine#2141] [Feature requests]
Hey guys,

at first I wan't to thank you for your great work. When I came over styling I personally missed the way of styling using the `@emtion/styled` package. It is working but I cannot access the real mantine `theme`.
From my point of view understanding software is better when dependencies are as closes as possible by what they affect. Therefore I like it more to use the `styled` method from emotion. 

My suggestion would be to add the `ThemeProvider` from `@emotion/react` to the `MantineProvider`.

If you think this should not be part of the `MantineProvider` can we add a function that returns me a full theme merged with my custom theme? Then I can add the `ThemeProvider` myself.

**Comment by rtivital** [comment]
Hi @jembach, thanks for a feature request, we will consider adding emotion styled support

**Reply by jembach** [reply]
@rtivital if you need help with implementation I will also create a PR.

**Reply by rtivital** [reply]
No, thanks, it will be better if I figure it out myself

**Reply by rtivital** [reply]
I've managed to get it working, it will be added as part of 5.2 release
